mlserver/parallel/worker.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import select
+
+from queue import Empty
+from multiprocessing import Process, Queue, JoinableQueue
+
+from ..registry import MultiModelRegistry
+
+from .messages import ModelUpdateType, ModelUpdateMessage, InferenceResponseMessage
+from .utils import terminate_queue, END_OF_QUEUE
+from .logging import logger
+
+
+class Worker(Process):
+    def __init__(self, requests: Queue, responses: Queue, model_updates: JoinableQueue):
+        super().__init__()
+        self._requests = requests
+        self._responses = responses
+        self.model_updates = model_updates
+
+    def run(self):
+        asyncio.run(self.coro_run())
+
+    def __inner_init__(self):
+        """
+        Internal __init__ method that needs to run within the worker process.
+        """
+        self._model_registry = MultiModelRegistry()
+        self._active = True
+
+    async def coro_run(self):
```

Comment:
is this private?

Reply:
No need to. It's just the async version of the `run()` method of the process. I think it's actually used by the tests (or it used to!).